src/components/CategoryList.js

```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from 'react';
+
+// eslint-disable-next-line import/order
+import { getFolderLink } from '../api/BaseUrl';
+
+import '../styles/CategoryList.css';
+
+import iconDelete from '../assets/logo/icon-delete.svg';
+import iconPen from '../assets/logo/icon-pen.svg';
+import iconShare from '../assets/logo/icon-share.svg';
+import iconPlus from '../assets/logo/plus.svg';
+
+import CardList from './CardList';
+import Category from './Category';
+
+const CategoryList = ({ folderInfo, categoryList, isIconVisible }) => {
+  const [selected, setSelected] = useState({ id: '', name: '전체' });
+  const [folderLink, setFolderLink] = useState(null);
+
+  const categoryData = categoryList?.data;
+
+  const handleLoad = async categoryId => {
+    try {
+      const state = await getFolderLink(categoryId);
+      setFolderLink(state);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    handleLoad(selected.id);
+  }, [selected.id]);
+
+  return (
+    <>
+      <div className='category-list'>
+        <div className='category-list-container'>
+          <a className={`category ${selected.name === '전체' ? 'category-selected' : ''}`} onClick={() => setSelected({ id: '', name: '전체' })}>
+            전체
+          </a>
+          {categoryData ? (
+            <>
+              {categoryData?.map(category => (
+                <Category key={category.id} category={category} onClickCategory={setSelected} isSelected={selected && selected.id === category.id} />
+              ))}
+            </>
+          ) : null}
+        </div>
+        <div className='add-category-container'>
+          <a className='add-category'>폴더 추가</a>
+          <img className='add-category-icon' src={iconPlus} alt='폴더 추가하기.' />
+        </div>
+      </div>
+
+      <div className='category-title'>
+        <h2 className='category-name'>{selected.name}</h2>
+        {selected.name !== '전체' ? (
+          <>
+            <div className='category-menu'>
+              <img className='category-menu-icon' src={iconShare} alt='폴더 공유하기.' />
+              <a className='category-menu-text'>공유</a>
+              <img className='category-menu-icon' src={iconPen} alt='폴더 이름 변경하기.' />
+              <a className='category-menu-text'>이름 변경</a>
+              <img className='category-menu-icon' src={iconDelete} alt='폴더 삭제하기.' />
+              <a className='category-menu-text'>삭제</a>
+            </div>
+          </>

```

Comment:
P3:
불필요한 태그는 지워주세요. 
이 부분 따로 컴포넌트로 빼시면 더 가독성이 좋을 것 같습니다.

```suggestion
            <div className='category-menu'>
              <img className='category-menu-icon' src={iconShare} alt='폴더 공유하기.' />
              <a className='category-menu-text'>공유</a>
              <img className='category-menu-icon' src={iconPen} alt='폴더 이름 변경하기.' />
              <a className='category-menu-text'>이름 변경</a>
              <img className='category-menu-icon' src={iconDelete} alt='폴더 삭제하기.' />
              <a className='category-menu-text'>삭제</a>
            </div>
```